### ISSUE TITLE
Typehinted properties on missing class with preloading

### ISSUE THREAD

**Issue by jderusse** [symfony/symfony#44364]
### Symfony version(s) affected

6.0

### Description

I'm using `Symfony\Bridge\Twig\Mime\BodyRenderer` without `League\HTMLToMarkdown\HtmlConverter`.
The code in the constructor allows me to do that.

But in production, with preloading, I'm facing an issue:

```
Failed to load class League\HTMLToMarkdown\HtmlConverter used by typed property Symfony\Bridge\Twig\Mime\BodyRenderer::$converter during preloading
```

I believe that typehinted property HAVE to be loadable when preloaded

edit: same for `Predis\ClientInterface` in `Symfony\Component\Cache\Traits\RedisClusterProxy`

### How to reproduce

- enable preloading
- `composer require symfony/twig-bridge:^6.0`
- assert `League\HTMLToMarkdown` is NOT installed

### Possible Solution

- remove type hinted property for optional dependencies

**Comment by nicolas-grekas** [comment]
> I believe that typehinted property HAVE to be loadable when preloaded

I guess so. @nikic is that correct?

> remove type hinted property for optional dependencies

Works for me. The alternative is to ignore these warnings.
Or maybe even to silence them: Unless you're actually working on increasing the ratio of preloaded classes, preload time warnings are not that useful. WDYT? Up to giving it a try?

**Comment by jderusse** [comment]
well, those are fatal error in my case. (maybe an error handler converted it I to exception somewhere, I had to downgraded to 5.4)

**Comment by nikic** [comment]
> > I believe that typehinted property HAVE to be loadable when preloaded
> 
> I guess so. @nikic is that correct?

Prior to PHP 8.1, yes. Since PHP 8.1 property types do not matter for preloading.

> Works for me. The alternative is to ignore these warnings. Or maybe even to silence them: Unless you're actually working on increasing the ratio of preloaded classes, preload time warnings are not that useful. WDYT? Up to giving it a try?

For `require`-based preloading these are fatal errors, otherwise warnings.

**Comment by nicolas-grekas** [comment]
> Prior to PHP 8.1, yes. Since PHP 8.1 property types do not matter for preloading.

Oh, good to know thanks!

> For require-based preloading these are fatal errors, otherwise warnings.

Then we have to revert all those typed props I suppose @jderusse, isn't it?

**Comment by nicolas-grekas** [comment]
Ideally we would even have a CI check for that. It's just not going to be trivial since it needs to combine info from the AST with info from composer files... Or does such a tool already exist somewhere? Anyone?

**Comment by nicolas-grekas** [comment]
Solved by bumping Symfony 6.1 to PHP 8.1, see https://github.com/symfony/symfony/pull/45377
